vm/starnix/starnix.go

```diff
@@ -248,13 +248,28 @@ func (inst *instance) connectToAdb(timeout time.Duration) error {
 }
 
 func (inst *instance) restartAdbAsRoot() error {
-	err := inst.runCommand(
-		"adb",
-		"-s",
-		fmt.Sprintf("127.0.0.1:%d", inst.port),
-		"root",
-	)
-	return err
+	startTime := time.Now()
+	for {
+		vmimpl.SleepInterruptible(3 * time.Second)
+		if inst.debug {
+			log.Logf(0, "attempting to restart ADB with root access")
```

Comment:
Nit: I'd use a lower log level for this (e.g. `1`) as it's really capable of littering syz-manager logs.

Reply:
Thanks, done. I amended this commit to set all of the per-retry logging to level 1, and made some small changes to the level-0 logging so that it's clear when a connection attempt is in progress.